keys.go

```diff
@@ -192,6 +206,40 @@ func (c *Client) GetKey(ctx context.Context, id string) (*Key, error) {
 	return &keys.Keys[0], nil
 }
 
+//GetKeyMetadata retrieves key metadata details by ID.
+func (c *Client) GetKeyMetadata(ctx context.Context, id string) (*Key, error) {
+	keys := Keys{}
+
+	req, err := c.newRequest("GET", fmt.Sprintf("keys/%s/metadata", id), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = c.do(ctx, req, &keys)
+	if err != nil {
+		return nil, err
+	}
+
+	return &keys.Keys[0], nil
+}
+
+//ListKeyVersions retrieves all the rotated key versions associated with the key
+func (c *Client) GetKeyVersions(ctx context.Context, id string) (*Key, error) {
+	keys := Keys{}
+
+	req, err := c.newRequest("GET", fmt.Sprintf("keys/%s/versions", id), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = c.do(ctx, req, &keys)
+	if err != nil {
+		return nil, err
+	}
+
+	return &keys.Keys[0], nil
```

Comment:
You won't be returning much here... since apparently you are just returning a single Key type with no version information...